### PR TITLE
Port tests to run for xs3

### PR DESCRIPTION
**Pull request by shuchitak**
- Run tests for xs3 architecture
- Support for generating random seed or provide a fixed seed on pytest cmd line when running tests 

Closes https://github.com/xmos/lib_ethernet/issues/56
Closes https://github.com/xmos/lib_ethernet/issues/39
Closes https://github.com/xmos/lib_ethernet/issues/57

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -80,26 +84,39 @@ pipeline {
...
-      		      sh "xmake -j 32 -C build"
-      		    } // script
-                runPytest('-vv')
-	      	  }
-	      	}
-      	  }
-      	}
-      }
-    }
+        sh "git clone git@github.com:xmos/test_support"
```
Comment (ed-xmos) [inline Jenkinsfile:87]: Don't we want to pin this?

File: lib_ethernet/src/mii_master.xc [hunk 2]
```diff
@@ -495,11 +495,11 @@ unsafe void mii_master_tx_pins(mii_mempool_t tx_mem_lp,
       credit_tmr :> credit_time;
 
       int elapsed = credit_time - prev_credit_time;
-      credit += elapsed * p_port_state->qav_idle_slope;
+      credit += elapsed * p_port_state->qav_idle_slope; // add bit budget since last transmission to credit. ticks * bits/tick = bits
```
Comment (ed-xmos) [inline lib_ethernet/src/mii_master.xc:498]: Thanks for adding comments

File: tests/mii_clock.py [hunk 3]
```diff
@@ -15,19 +15,22 @@ class Clock(px.SimThread):
     def __init__(self, port, clk):
         self._running = True
         self._clk = clk
-        sim_clock_rate = 1e15 # xsim uses femotseconds
+        sim_clock_rate = px.Xsi.get_xsi_tick_freq_hz() # xsim uses femotseconds
```
Comment (ed-xmos) [inline tests/mii_clock.py:18]: Cool

File: tests/test_rx_queues.py [hunk 4]
```diff
@@ -265,7 +264,7 @@ def test_rx_queues(capfd, params):
         (tx_clk_25, tx_mii) = get_mii_tx_clk_phy(test_ctrl='tile[0]:XS1_PORT_1C', expect_loopback=False, verbose=verbose)
 
         # Test having every packet going to both LP receivers
-        if params["test_id"] == "a":
+        if params["test_id"] == "hp_min_sz":
```
Comment (ed-xmos) [inline tests/test_rx_queues.py:267]: That's more readable!

File: tests/test_speed_change.py [hunk 5]
```diff
@@ -140,15 +138,20 @@ def create_expect(filename, packets_25, packets_125):
...
+def test_speed_change(capfd, seed, params):
+    if seed == None:
+        seed = random.randint(0, sys.maxsize)
+
+
     verbose = False
-    
+
     (tx_clk_25, tx_rgmii_25) = get_rgmii_tx_clk_phy(Clock.CLK_25MHz, initial_delay=initial_delay,
                                                   expect_loopback=False, verbose=verbose, dut_exit_time=5000000*1e6)
```
Comment (ed-xmos) [inline tests/test_speed_change.py:156]: Wonder if the DUT exit time could be more human readable? I think it's 5 milliseconds?

File: tests/test_tx.py [hunk 6]
```diff
@@ -63,29 +60,29 @@ def do_test(capfd, mac, arch, rx_clk, rx_phy, tx_clk, tx_phy):
...
+@pytest.mark.parametrize("params", generate_tests(test_params_file)[0], ids=generate_tests(test_params_file)[1])
 def test_tx(capfd, params):
     # Even though this is a TX-only test, both PHYs are needed in order to drive the mode pins for RGMII
-
     # Test 100 MBit - MII XS2
     if params["phy"] == "mii":
         (rx_clk_25, rx_mii) = get_mii_rx_clk_phy(packet_fn=packet_checker, test_ctrl='tile[0]:XS1_PORT_1C')
         (tx_clk_25, tx_mii) = get_mii_tx_clk_phy(do_timeout=False)
-        do_test(capfd, params["mac"], params["arch"], rx_clk_25, rx_mii, tx_clk_25, tx_mii)
+        do_test(capfd, params["mac"], params['arch'], rx_clk_25, rx_mii, tx_clk_25, tx_mii)
```
Comment (ed-xmos) [inline tests/test_tx.py:71]: Ha - is this the age old double vs single quote like 2 space tabs vs 4 space tabs, camel vs snake etc..?

**Review by ed-xmos — APPROVED** [review]
A monster PR with some very useful test fixes and structure improvements.
You seem to often be tidying up my work! First MIDI and now this..
Minor comments only so approving.

File: tests/test_speed_change.py (hunk 5 above)
Comment (shuchitak) [reply tests/test_speed_change.py:156]: I've fixed it.

**Pull request merged**